### DISCUSSION TITLE
Installing DFFML Module

### DISCUSSION THREAD

**Discussion by aryanxk02** [intel/dffml#1009] [Q&A]
I am new to this repository and finding some difficulty while installing this module. 
Help would be appreciated.

`aryan-ubuntu@aryan-ubuntu:~$ python3 -m dffml version`
`dffml 0.4.0 /home/aryan-ubuntu/.local/lib/python3.8/site-packages/dffml`
`dffml-config-yaml not installed`
`dffml-config-image not installed`
`dffml-model-scratch not installed`
`dffml-model-scikit not installed`
`dffml-model-tensorflow not installed`
`dffml-model-tensorflow-hub not installed`
`dffml-model-vowpalWabbit not installed`
`dffml-model-xgboost not installed`
`dffml-model-pytorch not installed`
`dffml-model-spacy not installed`
`dffml-model-daal4py not installed`
`dffml-model-autosklearn not installed`
`dffml-feature-git not installed`
`dffml-feature-auth not installed`
`dffml-operations-binsec not installed`
`dffml-operations-deploy not installed`
`dffml-operations-image not installed`
`dffml-operations-nlp not installed`
`dffml-service-http not installed`
`dffml-source-mysql not installed`
`aryan-ubuntu@aryan-ubuntu:~$ dffml version`
`dffml: command not found`

Thanks.

**Comment by aryanxk02** [comment]
i tried typing the command 
` python -m pip install -U dffml[models] "https://github.com/automl/ConfigSpace/archive/86863bc2f640892675c04de0b8006a623dc14c45.zip#egg=ConfigSpace"`

Error message occuring in Ubuntu 20.04 terminal -
`    Running setup.py install for pyrfr ... error
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-ruvy_iwb/pyrfr_a6c16d97de58458b830a60a9f2726787/setup.py'"'"'; __file__='"'"'/tmp/pip-install-ruvy_iwb/pyrfr_a6c16d97de58458b830a60a9f2726787/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-e6nj8feg/install-record.txt --single-version-externally-managed --user --prefix= --compile --install-headers /home/aryan-ubuntu/.local/include/python3.8/pyrfr
         cwd: /tmp/pip-install-ruvy_iwb/pyrfr_a6c16d97de58458b830a60a9f2726787/
    Complete output (7 lines):
    running install
    running build_ext
    building 'pyrfr._regression' extension
    swigging pyrfr/regression.i to pyrfr/regression_wrap.cpp
    swig -python -c++ -modern -features nondynamic -I./include -o pyrfr/regression_wrap.cpp pyrfr/regression.i
    unable to execute 'swig': No such file or directory
    error: command 'swig' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-ruvy_iwb/pyrfr_a6c16d97de58458b830a60a9f2726787/setup.py'"'"'; __file__='"'"'/tmp/pip-install-ruvy_iwb/pyrfr_a6c16d97de58458b830a60a9f2726787/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-e6nj8feg/install-record.txt --single-version-externally-managed --user --prefix= --compile --install-headers /home/aryan-ubuntu/.local/include/python3.8/pyrfr Check the logs for full command output.
`
What's the matter?

**Reply by johnandersen777** [reply]
You probably need to check out https://automl.github.io/auto-sklearn/master/installation.html#ubuntu-installation

**Comment by aryanxk02** [comment]
okay I have figured it out!

**Comment by johnandersen777** [comment]
If you try to run DFFML without the module invocation and get a "command not found" error, you need to add it to your `PATH` environment variable:

```console
aryan-ubuntu@aryan-ubuntu:~$ dffml version
dffml: command not found
```

When you run the version command you see the location each package is installed to:

> ```console
> aryan-ubuntu@aryan-ubuntu:~$ python3 -m dffml version
> dffml 0.4.0 /home/aryan-ubuntu/.local/lib/python3.8/site-packages/dffml
> ```

When packages are installed to `~/.local/lib/pythonX.Y/site-packages/`, their scripts are installed to `~/.local/bin`

Try adding the following to your `~/.bashrc` or `~/.bash_profile`:

```bash
export PATH="${HOME}/.local/bin:${PATH}"
```

Also, run that line in a terminal
